src/pages/staticPages/tools/oncoprinter/OncoprinterStore.ts

```diff
@@ -0,0 +1,311 @@
+        } else {
+            return undefined;
+        }
+    }
+
+    @action setGeneOrder(input:string) {
+        this._geneOrder = input.trim();
+    }
+
+    @computed get geneOrder() {
+        if (this._geneOrder) {
+            return _.uniq(this._geneOrder.split(/[,\s]+/));
+        } else {
+            return undefined;
+        }
+    }
+
+    @action setDataInput(input:string) {
+        this._dataInput = input;
+    }
+
+    @action public setInput(data:string, genes:string, samples:string) {
+        this.setDataInput(data);
+        this.setGeneOrder(genes);
+        this.setSampleIdOrder(samples);
+    }
+
+    public readonly parsedInputLines = remoteData({
+        invoke:async()=>{
+            if (!this._dataInput) {
+                return [];
+            }
+            // parse input
+            const parsed = parseInput(this._dataInput);
+            if (parsed.status === "error") {
+                throw new Error(parsed.error);
+            } else {
+                let parsedLines = parsed.result;
+                // fetch hugoGeneSymbols and hotspot for Type3 input lines
+                const toAnnotate =
+                    _.chain(parsedLines as any).filter(l=>isType3NoGene(l)).uniqBy(genomeNexusKey).values().value();
+                if (toAnnotate.length > 0) {
+                    try {
+                        // get data and store in map for querying
+                        const gnData = _.keyBy(await gnClient.fetchVariantAnnotationByGenomicLocationPOST({
+                            genomicLocations: toAnnotate.map(line=>({
+                                chromosome: line.chromosome,
+                                start: line.startPosition,
+                                end: line.endPosition,
+                                referenceAllele: line.referenceAllele,
+                                variantAllele: line.variantAllele
+                            })),
+                            fields:["hotspots", "annotation_summary"]
+                        }), d=>genomeNexusKey2(d.annotation_summary.genomicLocation));
+                        // annotate lines in place - this keeps their place in parsedLines in original data order
+                        for (const line of toAnnotate) {
+                            const gnDatum = gnData[genomeNexusKey(line)];
+                            (line as OncoprinterInputLineType3).hugoGeneSymbol = gnDatum.annotation_summary.transcriptConsequenceSummary.hugoGeneSymbol;
+                            (line as OncoprinterInputLineType3).isHotspot = _.flatten(gnDatum.hotspots.annotation).filter(recurrentHotspotFilter).length > 0;
+                        }
+                    } catch (e) {
+                        // GenomeNexus error - filter out type3 lines because we don't know how to use them
+                        alert("Error fetching data from GenomeNexus service - throwing out Type 3 (MAF/mutation) input data rows.");
```

Comment:
maybe good time to talk about how we can handle alerts

Reply:
lets talk about it for next round